services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/entity/EntityAuthorizationManagerAutowireTest.java

```diff
@@ -370,17 +457,464 @@ public void testHasAccessDownloadWithUnmetAccessRestrictions() {
+		});
+
+		// userTwo is contributor for fileTwo with no download access
+		aclHelper.create((a) -> {
+			a.setId(fileTwo.getId());
+			a.setResourceAccess(Collections.singleton(
+					new ResourceAccess().setPrincipalId(userTwo.getId())
+							.setAccessType(Set.of(ACCESS_TYPE.UPDATE, ACCESS_TYPE.DELETE))));
+		});
+
+		// userTwo is contributor for fileThree with no download access
+		aclHelper.create((a) -> {
+			a.setId(fileThree.getId());
+			a.setResourceAccess(Collections.singleton(
+					new ResourceAccess().setPrincipalId(userTwo.getId())
+							.setAccessType(Set.of(ACCESS_TYPE.UPDATE, ACCESS_TYPE.DELETE))));
+		});
+
+		//AR on fileOne and fileTwo
+		Long arId = managedHelper.create(a -> {
+			a.setAccessType(ACCESS_TYPE.DOWNLOAD);
+			a.setSubjectIds(List.of(
+					new RestrictableObjectDescriptor()
+							.setId(fileOne.getId())
+							.setType(RestrictableObjectType.ENTITY),
+					new RestrictableObjectDescriptor()
+							.setId(fileTwo.getId())
+							.setType(RestrictableObjectType.ENTITY)));
+		}).getId();
+
+		//Exemption ACL on arId
+		aclHelper.create((a) -> {
+			a.setId(arId.toString());
+			a.setResourceAccess(Set.of(
+					new ResourceAccess()
+							.setPrincipalId(Long.valueOf(userTwo.getId()))
+							.setAccessType(Collections.singleton(ACCESS_TYPE.EXEMPTION_ELIGIBLE))));
+		} , ObjectType.ACCESS_REQUIREMENT);
+
+		// There is no AR on file three and also no download access
 		String newMessage = assertThrows(UnauthorizedException.class, () -> {
-			entityAuthManager.hasAccess(user, entityId, accessType).checkAuthorizationOrElseThrow();
+			entityAuthManager.hasAccess(userTwo, fileThree.getId(), ACCESS_TYPE.DOWNLOAD).checkAuthorizationOrElseThrow();
+		}).getMessage();
+		assertEquals("You lack DOWNLOAD access to the requested entity.", newMessage);
+
+		// userTwo is exempted for file Two but not met access requirement and has no download access.
+		String newMessageTwo = assertThrows(UnauthorizedException.class, () -> {
+			entityAuthManager.hasAccess(userTwo, fileTwo.getId(), ACCESS_TYPE.DOWNLOAD).checkAuthorizationOrElseThrow();
 		}).getMessage();
-		assertEquals(ERR_MSG_THERE_ARE_UNMET_ACCESS_REQUIREMENTS, newMessage);
+		assertEquals("You lack DOWNLOAD access to the requested entity.", newMessageTwo);
+
+		//userTwo is contributor for file One and exempted
+		AuthorizationStatus status = entityAuthManager.hasAccess(userTwo, fileOne.getId(), ACCESS_TYPE.DOWNLOAD);
+		assertNotNull(status);
+		assertTrue(status.isAuthorized());
+
+		//userOne is neither exempted nor met access approval
+		String message = assertThrows(UnauthorizedException.class, () -> {
+			entityAuthManager.hasAccess(userOne, fileOne.getId(), ACCESS_TYPE.DOWNLOAD).checkAuthorizationOrElseThrow();
+		}).getMessage();
+		assertEquals("There are unmet access requirements that must be met and exemption is not eligible to read content in the requested container.", message);
```

Comment:
message should just be they have unmet AR.  We are not trying to inform them about the exemption.